pkg/internal/transform/kube/db.go

```diff
@@ -167,34 +150,57 @@ func (id *Database) OwnerPodInfo(pidNamespace uint32) (*kube.PodInfo, bool) {
 
 func (id *Database) UpdateNewPodsByIPIndex(pod *kube.PodInfo) {
 	if len(pod.IPInfo.IPs) > 0 {
-		id.podsMut.Lock()
-		defer id.podsMut.Unlock()
-		for _, ip := range pod.IPInfo.IPs {
-			id.podsByIP[ip] = pod
-		}
+		id.access.Lock()
```

Comment:
I think this can lead to a race too, we check the pod state before we lock. Let's wrap the whole function in a lock.

Reply:
In this case, it's not needed because the Mutex protects access to the maps, and the `pod` value is provided from outside the database, from a value that is not concurrently modified.

Anyway let's move the lock outside the block to make it clearer, as anyway the pods should have always IPs.